lib/wallaby/browser.ex

```diff
@@ -903,6 +903,65 @@ defmodule Wallaby.Browser do
   end
 
   @doc """
+  Checks if `query` is present within `parent` and raises if not found.
+
+  Returns the given `parent` if the assertion is correct so that it is easily
+  pipeable.
+
+  ## Examples
+
+      session
+      |> visit("/")
+      |> assert_has(Query.css(".login-button"))
+  """
+  @spec assert_has(parent, Query.t) :: parent
+
+  defmacro assert_has(parent, query) do
+    quote do
+      parent = unquote(parent)
+      query  = unquote(query)
+
+      case has?(parent, query) do
+        true ->
+          parent
+        false ->
+          raise Wallaby.ExpectationNotMet,
+                Query.ErrorMessage.message(query, :not_found)
+
+      end
+    end
+  end
+
+  @doc """
+  Checks if `query` is not present within `parent` and raises if it is found.
+
+  Returns the given `parent` if the query is not found so that it is easily
+  pipeable.
+
+  ## Examples
+
+      session
+      |> visit("/")
+      |> refute_has(Query.css(".secret-admin-content"))
+  """
+  @spec refute_has(parent, Query.t) :: parent
+
+  defmacro refute_has(parent, query) do
+    quote do
+      parent = unquote(parent)
+      query  = unquote(query)
+
+      case execute_query(parent, query) do
+        {:error, _not_found} ->
+          parent
+        {:ok, query} ->
```

Comment:
Ah ok, I think I see the issue. Looks like we will need to use `execute_query` for both `assert_has` and `refute_has`. The error for `execute_query` should be something like `{:not_found, results}`. So you should be able to get the result value out of the error in `assert_has`. If not then there is probably a different bug somewhere.